cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -2347,3 +2340,191 @@ where
         Ok(func)
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::OPCODE_SIGNATURES;
+    use cranelift::codegen::ir::{instructions::ResolvedConstraint, types::*, Opcode, Type};
+    use std::collections::HashMap;
+
+    /// The list of opcodes that we exclude from the analysis. This should only be instructions
+    /// that affect control flow, as everything else should be handled by the general machinery in
+    /// function_generator.
+    ///
+    /// NOTE: This list should not grow, and ultimately should just be the set of instructions that
+    /// need special handling outside of `generate_instructions`.
+    const SKIPPED_OPCODES: &[Opcode] = &[
+        // These opcodes have special handling in cranelift-fuzzgen
+        Opcode::Call,
+        Opcode::Return,
+        Opcode::Jump,
+        Opcode::Brif,
+        Opcode::BrTable,
+        // TODO: ExtractVector produces dynamic vectors, and those cause a panic in
+        // `constraint.result_type`.
+        Opcode::ExtractVector,
+        // These opcodes are known to have invalid signatures, but the specific types chosen are
+        // not used by the insertion strategy for that format. They can be removed when their
+        // entries in OPCODE_SIGNATURES have been fixed
+        Opcode::Load,
+        Opcode::Store,
+        Opcode::AtomicLoad,
+        Opcode::AtomicStore,
+        Opcode::AtomicCas,
+        Opcode::AtomicRmw,
+        Opcode::Sload8,
+        Opcode::Sload16,
+        Opcode::Sload32,
+        Opcode::Uload8,
+        Opcode::Uload16,
+        Opcode::Uload32,
+        Opcode::Istore8,
+        Opcode::Istore16,
+        Opcode::Istore32,
```

Comment:
This is the current list of discrepancies that we'll need to fix to ensure that `OPCODE_SIGNATURES` is reflecting the current state of instruction constraints. A separate issue will be to reconcile the instruction constraints with the current behavior of the backends, but we'll get there eventually :)